sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -553,6 +553,40 @@ def test_ordinal_encoder_raise_missing(X):
         ohe.transform(X)
 
 
+def test_ordinal_encoder_handle_unknowns():
+    enc = OrdinalEncoder(unknown_value=-999)
+    X_fit = np.array([['a', 'x'], ['b', 'y'], ['c', 'z']], dtype=object)
+    X_trans = np.array([['c', 'xy'], ['bla', 'y'], ['a', 'x']], dtype=object)
+    enc.fit(X_fit)
+
+    X_trans_enc = enc.transform(X_trans)
+    exp = np.array([[2, -999], [-999, 1], [0, 0]], dtype=object)
```

Comment:
The dtype of the output should not be object. Is this the case here?

Reply:
right, it is integer.